README.md
# SDA5708 Arduino Library

The company Siemens produced these tiny 8 character 5x7 dot matrix displays. They were widely used in some Nokia Receivers like the DBOX1 in Germany. I don't know if there are other sources but as I have so many of these displays I decided to make an Arduino Library to easily use them in small projects.

With this library you can use the dot matrix display similar to LCD Displays. Either multiple independant displays with multiple instances of the class or multiple displays combined together with 16, 24, 32, ... characters in a row. 

Please refer to the documentation provided.


changes.txt

SDA5708 dot matrix Library by Andreas Rohmann, licensed into public domain.


HISTORY

Aug 21, 2022 - initial release

SDA5708.h
/*
  SDA5708.H – Library for using SDA5708 Display(s) from old Nokia DBOX1 Receivers
  Created by Andreas Rohmann, released into public domain.
*/

#ifndef __SDA5708__
#define __SDA5708__

#if (ARDUINO>=100)
    #include "Arduino.h"
#else
    #include "WProgram.h"
#endif

#include "font.h"

// constants
#define SDA5708_BRIGHTNESS_MAX  0
#define SDA5708_BRIGHTNESS_MIN  6
#define SDA5708_BRIGHTNESS_OFF  7

class SDA5708 : public Print {
  
	public:
	
	// constructors
    SDA5708(uint8_t pin_reset, uint8_t pin_data, uint8_t pin_clock, uint8_t pin_load);
    SDA5708(uint8_t no_of_displays, uint8_t pin_reset, uint8_t pin_data, uint8_t pin_clock, uint8_t pin_load);

	// methods
	void reset();
	void clear();
	void setBrightness(uint8_t bright);
	void setCursor(uint8_t pos);

    size_t write(uint8_t chr, uint8_t pos);
    size_t write(uint8_t chr);

	size_t print(const __FlashStringHelper *);
	size_t print(const String &);
	size_t print(const char[]);
	size_t print(char);
	size_t print(unsigned char, int = DEC);
	size_t print(int, int = DEC);
	size_t print(unsigned int, int = DEC);
	size_t print(long, int = DEC);
	size_t print(unsigned long, int = DEC);
	size_t print(double, int = 2);
	size_t print(const Printable&);
	
	size_t println(const __FlashStringHelper *);
	size_t println(const String &s);
	size_t println(const char[]);
	size_t println(char);
	size_t println(unsigned char, int = DEC);
	size_t println(int, int = DEC);
	size_t println(unsigned int, int = DEC);
	size_t println(long, int = DEC);
	size_t println(unsigned long, int = DEC);
	size_t println(double, int = 2);
	size_t println(const Printable&);
	size_t println(void);

	// optional methods
	#ifndef SDA5708_NO_UDC
  		void createChar(uint8_t no, uint8_t r0, uint8_t r1, uint8_t r2, uint8_t r3, uint8_t r4, uint8_t r5, uint8_t r6);
	#endif

  private:

    // internal variables
    uint8_t _no_of_displays = 1;  // displays to control, default for 1 display
    uint8_t _no_of_digits = 8;    // available digits, default for 1 display
    uint8_t _pin_load;            // load pin of first display
    uint8_t _pin_data;            // common data pin for data transfer
    uint8_t _pin_clock;           // common clock pin for data transfer
    uint8_t _pin_reset;           // common display reset signal
    uint8_t _position;            // internal write position

    // internal methods
    void send_byte(uint8_t display_no, uint8_t b);
    void set_ports();
    
	// optional internal variables
	#ifndef SDA5708_NO_UDC 
  		char udc[16*7];
	#endif

};

// --------------------------------------------------------------------------------
// All methods need to be integrated into the header file to be compiled with
// the Arduino sketch. Otherwise defines wouldn't be recognized during compile time.
// Normally these methods would be in a separate .CPP file. 
// One of the disadvantages by using the Arduino IDE :-(
// --------------------------------------------------------------------------------

/*
 * Constructor of this class for using a single SDA5708 display.
 */
SDA5708::SDA5708(uint8_t pin_reset, uint8_t pin_data, uint8_t pin_clock, uint8_t pin_load) 
{
    _pin_reset = pin_reset;
    _pin_data = pin_data;
    _pin_clock = pin_clock;
    _pin_load = pin_load;
    set_ports();
    reset();
}

/*
 * Constructor of this class for using multiple SDA5708 displays.
 */
SDA5708::SDA5708(uint8_t no_of_displays, uint8_t pin_reset, uint8_t pin_data, uint8_t pin_clock, uint8_t pin_load) 
{
    _no_of_displays = no_of_displays;
    _no_of_digits = 8*no_of_displays;
    _pin_reset = pin_reset;
    _pin_data = pin_data;
    _pin_clock = pin_clock;
    _pin_load = pin_load;
    set_ports();
    reset();
}


/*
 * Internal method to initialize all used output pins.
 */
void SDA5708::set_ports() 
{
    pinMode(_pin_reset, OUTPUT);
    pinMode(_pin_data, OUTPUT);
    pinMode(_pin_clock, OUTPUT);
    for (uint8_t x = 0; x < _no_of_displays; x++) {
        pinMode(_pin_load + x, OUTPUT);
    }
}


/*
 * This method performs a hardware reset of all connected displays.
 * The internal write position is set to 0.
 */
void SDA5708::reset() 
{
    for (uint8_t x = 0; x < _no_of_displays; x++) {
        digitalWrite(_pin_load + x, HIGH);
        digitalWrite(_pin_reset, LOW);
        digitalWrite(_pin_reset, HIGH);
    }
    _position = 0;
}


/*
 *  Internal method to send one byte to a particular display.
 */
void SDA5708::send_byte(uint8_t display_no, uint8_t data) 
{
    // enable display
    digitalWrite(_pin_load + display_no, LOW);
    // send 8 bits
    for (uint8_t x = 0; x <= 7; x++) {
        // set data pin 
        digitalWrite(_pin_data, ((data >> x) & 0x01));
        // toggle SDCLK 
        digitalWrite(_pin_clock, HIGH);
        digitalWrite(_pin_clock, LOW);
    }
    // disable display
    digitalWrite(_pin_load + display_no, HIGH);
}


/*
 * This method sets the cursor (nect writing position)
 * to the given value. SDA displays are always treated 
 * as single line displays.
 */
void SDA5708::setCursor(uint8_t pos)
{
    _position = pos;
}

/*
 * This method clears all registered displays 
 * and sets the internal write position to 0. 
 */
void SDA5708::clear() 
{
    for (uint8_t x = 0; x < _no_of_displays; x++) {
        send_byte(x, B11000000);
        send_byte(x, B11100000);
    }
    _position = 0;
}


/*
 * This method sets the brightness for all characters 
 * of all registered displays. Values are from 0 to 7: 
 * 0=maximum brighnexx, 6=minimum brightness, 7=off
 */
void SDA5708::setBrightness(uint8_t bright) 
{
    for (uint8_t x = 0; x < _no_of_displays; x++) {
        send_byte(x, B11100000 | (bright & B00000111));
    }
}




/*
 * This method writes a character at a given position.
 * If the position is out of range nothing happens.
 * After writing the character the internal write 
 * position points to the next digit. 
 */
size_t SDA5708::write(uint8_t chr, uint8_t pos) {
    // safety, write position in visible range
    if (pos >= _no_of_digits) return;
    // determine display to write to
    uint8_t disp = pos>>3;
    // set adress register (position)
    send_byte(disp, B10100000 | (pos & B00000111));
    // limit character range
    if ((chr >= SDA5708_FONT_START) && (chr <= SDA5708_FONT_END)) {
        // send character from font
        for (uint8_t i=0; i<7; i++) {
            send_byte(disp, pgm_read_byte(&font[(chr-SDA5708_FONT_START)*7+i]));
        }
    }
    // optional user defined characters
    #ifndef SDA5708_NO_UDC 
    else if ((chr >= 0xF0) && (chr <= 0xFF)) {
        // send character from udc
        for (uint8_t i=0; i<7; i++) {
            send_byte(disp, udc[(chr-0xF0)*7+i]);
        }
    }
    #endif // #ifndef SDA5708_NO_UDC
    _position++;
    return 1;
}


/*
 * This method writes a character at the internal write position.
 * After writing the character the internal write 
 * position points to the next digit.
 */
size_t SDA5708::write(uint8_t chr) {
    return write(chr, _position);
}



// suppport for basic printing 

size_t SDA5708::print(const __FlashStringHelper *fsh) 
{
    Print::print(fsh);
}

size_t SDA5708::print(const String& string)
{
    Print::print(string);
}

size_t SDA5708::print(const char string[])
{
    Print::print(string);
}

size_t SDA5708::print(char c)
{
    Print::print(c);
}

size_t SDA5708::print(unsigned char c, int base)
{
    Print::print(c, base);
}

size_t SDA5708::print(int i, int base)
{
    Print::print(i, base);
}

size_t SDA5708::print(unsigned int ui, int base)
{
    Print::print(ui, base);
}

size_t SDA5708::print(long l, int base)
{
    Print::print(l, base);
}

size_t SDA5708::print(unsigned long ul, int base)
{
    Print::print(ul, base);
}

size_t SDA5708::print(double num, int digits)
{
    Print::print(num, digits);
}

size_t SDA5708::print(const Printable& printable)
{
    Print::print(printable)	;
}

/*
 * Positioning of cursor happens before printing.
 * println() means that printing starts at position 0.
 */

size_t SDA5708::println(const __FlashStringHelper* fsh)
{
    _position = 0;
    Print::print(fsh);
}

size_t SDA5708::println(const String &s)
{
    _position = 0;
    Print::print(s);
}

size_t SDA5708::println(const char string[])
{
    _position = 0;
    Print::print(string);
}

size_t SDA5708::println(char c)
{
    _position = 0;
    Print::print(c);
}

size_t SDA5708::println(unsigned char b, int base)
{
    _position = 0;
    Print::print(b, base);
}

size_t SDA5708::println(int num, int base)
{
    _position = 0;
    Print::print(num, base);
}

size_t SDA5708::println(unsigned int num, int base)
{
    _position = 0;
    Print::print(num, base);
}

size_t SDA5708::println(long num, int base)
{
    _position = 0;
    Print::print(num, base);
}

size_t SDA5708::println(unsigned long num, int base)
{
    _position = 0;
    Print::print(num, base);
}

size_t SDA5708::println(double num, int digits)
{
    _position = 0;
    Print::print(num, digits);
}

size_t SDA5708::println(const Printable& printable)
{
    _position = 0;
    Print::print(printable);
}

size_t SDA5708::println(void)
{
    _position = 0;
}



#ifndef SDA5708_NO_UDC
/*
 * This method creates a user defined character.
 * User defined charcters must be in the range from 0xF0 to 0xFF
 */
void SDA5708::createChar(uint8_t no, uint8_t r0, uint8_t r1, uint8_t r2, uint8_t r3, uint8_t r4, uint8_t r5, uint8_t r6) {
    if (no>=0xF0 & no<=0xFF) {
        no -= 0xF0;
        udc[7*no+0] = r0;
        udc[7*no+1] = r1;
        udc[7*no+2] = r2;
        udc[7*no+3] = r3;
        udc[7*no+4] = r4;
        udc[7*no+5] = r5;
        udc[7*no+6] = r6;
    }
}
#endif // #ifndef SDA5708_NO_UDC



#endif // #ifndef __SDA5708__


font.h
/*
  FONT.H - for use with SDA5708 Library
  Created by Andreas Rohmann, released into public domain
  
  TODO
  
  - Mathematic Symbols
  - Electronic Symbols
*/

#ifndef __SDA5708_FONT__
#define __SDA5708_FONT__

#if (ARDUINO>=100)
  #include "Arduino.h"
#else
  #include "WProgram.h"
#endif

#define SDA5708_FONT_START  0x20

#ifdef SDA5708_EXCLUDE_SPECIAL
  #define _special 0
#else
  #define _special 0x10
#endif

#ifdef SDA5708_INCLUDE_ARROWS
  #define _arrows 0x10
#else
  #define _arrows 0
#endif

#ifdef SDA5708_INCLUDE_HBLOCKS
  #define _hblocks 0x10
#else
  #define _hblocks 0
#endif

#ifdef SDA5708_INCLUDE_VBLOCKS
  #define _vblocks 0x10
#else
  #define _vblocks 0
#endif

#ifdef SDA5708_INCLUDE_GRAPHICS
  #define _graphics 0x10
#else
  #define _graphics 0
#endif

#define SDA5708_FONT_END    (0x7F + _special + _arrows + _hblocks + _vblocks + _graphics)
 
#define SDA5708_FONT_LENGTH (SDA5708_FONT_END - SDA5708_FONT_START + 1)
#define SDA5708_FONT_BYTES  (SDA5708_FONT_LENGTH * 7)


const unsigned char font[SDA5708_FONT_BYTES] PROGMEM = {

   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,      // 20 SPACE
   0x04, 0x04, 0x04, 0x04, 0x04, 0x00, 0x04,      // 21 !
   0x0A, 0x0A, 0x00, 0x00, 0x00, 0x00, 0x00,      // 22 „
   0x0A, 0x0A, 0x1F, 0x0A, 0x1F, 0x0A, 0x0A,      // 23 #
   0x04, 0x0F, 0x14, 0x0E, 0x05, 0x1E, 0x04,      // 24 $
   0x18, 0x19, 0x02, 0x04, 0x08, 0x13, 0x03,      // 25 %
   0x0C, 0x12, 0x14, 0x08, 0x15, 0x12, 0x0D,      // 26 &
   0x0C, 0x04, 0x08, 0x00, 0x00, 0x00, 0x00,      // 27 '
   0x02, 0x04, 0x08, 0x08, 0x08, 0x04, 0x02,      // 28 (
   0x08, 0x04, 0x02, 0x02, 0x02, 0x04, 0x08,      // 29 )
   0x00, 0x04, 0x15, 0x0E, 0x15, 0x04, 0x00,      // 2A *
   0x00, 0x04, 0x04, 0x1F, 0x04, 0x04, 0x00,      // 2B +
   0x00, 0x00, 0x00, 0x00, 0x0C, 0x04, 0x08,      // 2C ,
   0x00, 0x00, 0x00, 0x1F, 0x00, 0x00, 0x00,      // 2D -
   0x00, 0x00, 0x00, 0x00, 0x00, 0x0C, 0x0C,      // 2E .
   0x01, 0x01, 0x02, 0x04, 0x08, 0x10, 0x10,      // 2F /

   0x0E, 0x11, 0x11, 0x15, 0x11, 0x11, 0x0E,      // 30 0
   0x02, 0x06, 0x0A, 0x02, 0x02, 0x02, 0x07,      // 31 1
   0x0E, 0x11, 0x01, 0x06, 0x08, 0x11, 0x1F,      // 32 2
   0x0E, 0x11, 0x01, 0x06, 0x01, 0x11, 0x0E,      // 33 3
   0x10, 0x10, 0x12, 0x1F, 0x02, 0x02, 0x02,      // 34 4
   0x1F, 0x10, 0x1E, 0x01, 0x01, 0x11, 0x0E,      // 35 5
   0x06, 0x08, 0x10, 0x1E, 0x11, 0x11, 0x0E,      // 36 6
   0x1F, 0x11, 0x02, 0x04, 0x08, 0x08, 0x08,      // 37 7
   0x0E, 0x11, 0x11, 0x0E, 0x11, 0x11, 0x0E,      // 38 8
   0x0E, 0x11, 0x11, 0x0F, 0x01, 0x11, 0x0E,      // 39 9
   0x00, 0x0C, 0x0C, 0x00, 0x0C, 0x0C, 0x00,      // 3A :
   0x00, 0x0C, 0x0C, 0x00, 0x0C, 0x04, 0x08,      // 3B ;
   0x02, 0x04, 0x08, 0x10, 0x08, 0x04, 0x02,      // 3C <
   0x00, 0x00, 0x1F, 0x00, 0x1F, 0x00, 0x00,      // 3D =
   0x08, 0x04, 0x02, 0x01, 0x02, 0x04, 0x08,      // 3E >
   0x0E, 0x11, 0x01, 0x02, 0x04, 0x00, 0x04,      // 3F ?

   0x0E, 0x11, 0x01, 0x0D, 0x15, 0x15, 0x0E,      // 40 @
   0x0E, 0x11, 0x11, 0x11, 0x1F, 0x11, 0x11,      // 41 A
   0x1E, 0x11, 0x11, 0x1E, 0x11, 0x11, 0x1E,      // 42 B
   0x0E, 0x11, 0x10, 0x10, 0x10, 0x11, 0x0E,      // 43 C
   0x1C, 0x12, 0x11, 0x11, 0x11, 0x12, 0x1C,      // 44 D
   0x1F, 0x10, 0x10, 0x1E, 0x10, 0x10, 0x1F,      // 45 E
   0x1F, 0x10, 0x10, 0x1E, 0x10, 0x10, 0x10,      // 46 F
   0x0E, 0x11, 0x10, 0x17, 0x11, 0x11, 0x0F,      // 47 G
   0x11, 0x11, 0x11, 0x1F, 0x11, 0x11, 0x11,      // 48 H
   0x0E, 0x04, 0x04, 0x04, 0x04, 0x04, 0x0E,      // 49 I
   0x07, 0x01, 0x01, 0x01, 0x01, 0x09, 0x06,      // 4A J
   0x11, 0x12, 0x14, 0x18, 0x14, 0x12, 0x11,      // 4B K
   0x10, 0x10, 0x10, 0x10, 0x10, 0x10, 0x1F,      // 4C L
   0x11, 0x1B, 0x15, 0x15, 0x11, 0x11, 0x11,      // 4D M
   0x11, 0x11, 0x19, 0x15, 0x13, 0x11, 0x11,      // 4E N
   0x0E, 0x11, 0x11, 0x11, 0x11, 0x11, 0x0E,      // 4F O

   0x1E, 0x11, 0x11, 0x1E, 0x10, 0x10, 0x10,      // 50 P
   0x0E, 0x11, 0x11, 0x11, 0x15, 0x12, 0x0D,      // 51 Q
   0x1E, 0x11, 0x11, 0x1E, 0x14, 0x12, 0x11,      // 52 R
   0x0F, 0x10, 0x10, 0x0E, 0x01, 0x01, 0x1E,      // 53 S
   0x1F, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04,      // 54 T
   0x11, 0x11, 0x11, 0x11, 0x11, 0x11, 0x0E,      // 55 U
   0x11, 0x11, 0x11, 0x11, 0x11, 0x0A, 0x04,      // 56 V
   0x11, 0x11, 0x11, 0x15, 0x15, 0x15, 0x0A,      // 57 W	
   0x11, 0x11, 0x0A, 0x04, 0x0A, 0x11, 0x11,      // 58 X
   0x11, 0x11, 0x11, 0x0A, 0x04, 0x04, 0x04,      // 59 Y
   0x1F, 0x01, 0x02, 0x04, 0x08, 0x10, 0x1F,      // 5A Z
   0x0E, 0x08, 0x08, 0x08, 0x08, 0x08, 0x0E,      // 5B [
   0x10, 0x10, 0x08, 0x04, 0x02, 0x01, 0x01,      // 5C backslash
   0x0E, 0x02, 0x02, 0x02, 0x02, 0x02, 0x0E,      // 5D ]
   0x04, 0x0A, 0x11, 0x00, 0x00, 0x00, 0x00,      // 5E ^
   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x1F,      // 5F _

   0x08, 0x04, 0x02, 0x00, 0x00, 0x00, 0x00,      // 60 `
   0x00, 0x00, 0x0E, 0x01, 0x0F, 0x11, 0x0F,      // 61 a
   0x10, 0x10, 0x16, 0x19, 0x11, 0x11, 0x1E,      // 62 b
   0x00, 0x00, 0x0E, 0x11, 0x10, 0x11, 0x0E,      // 63 c
   0x01, 0x01, 0x0D, 0x13, 0x11, 0x11, 0x0F,      // 64 d
   0x00, 0x00, 0x0E, 0x11, 0x1F, 0x10, 0x0E,      // 65 e
   0x06, 0x09, 0x08, 0x1C, 0x08, 0x08, 0x08,      // 66 f
   0x00, 0x00, 0x0F, 0x11, 0x0F, 0x01, 0x0E,      // 67 g
   0x10, 0x10, 0x16, 0x19, 0x11, 0x11, 0x11,      // 68 h	
   0x04, 0x00, 0x0C, 0x04, 0x04, 0x04, 0x0E,      // 69 i
   0x02, 0x00, 0x06, 0x02, 0x02, 0x12, 0x0C,      // 6A j
   0x10, 0x10, 0x12, 0x14, 0x18, 0x14, 0x12,      // 6B k
   0x0C, 0x04, 0x04, 0x04, 0x04, 0x04, 0x0E,      // 6C l
   0x00, 0x00, 0x1A, 0x15, 0x15, 0x11, 0x11,      // 6D m
   0x00, 0x00, 0x16, 0x19, 0x11, 0x11, 0x11,      // 6E n
   0x00, 0x00, 0x0E, 0x11, 0x11, 0x11, 0x0E,      // 6F o

   0x00, 0x00, 0x1E, 0x11, 0x1E, 0x10, 0x10,      // 70 p
   0x00, 0x00, 0x0D, 0x13, 0x0F, 0x01, 0x01,      // 71 q
   0x00, 0x00, 0x16, 0x19, 0x10, 0x10, 0x10,      // 72 r
   0x00, 0x00, 0x0E, 0x10, 0x0E, 0x01, 0x1E,      // 73 s
   0x08, 0x08, 0x1C, 0x08, 0x08, 0x09, 0x06,      // 74 t
   0x00, 0x00, 0x11, 0x11, 0x11, 0x13, 0x0D,      // 75 u
   0x00, 0x00, 0x11, 0x11, 0x11, 0x0A, 0x04,      // 76 v
   0x00, 0x00, 0x11, 0x11, 0x15, 0x15, 0x0A,      // 77 w
   0x00, 0x00, 0x11, 0x0A, 0x04, 0x0A, 0x11,      // 78 x
   0x00, 0x00, 0x11, 0x11, 0x0F, 0x01, 0x0E,      // 79 y
   0x00, 0x00, 0x1F, 0x02, 0x04, 0x08, 0x1F,      // 7A z
   0x02, 0x04, 0x04, 0x08, 0x04, 0x04, 0x02,      // 7B {
   0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04,      // 7C |
   0x08, 0x04, 0x04, 0x02, 0x04, 0x04, 0x08,      // 7D }
   0x0D, 0x16, 0x00, 0x00, 0x00, 0x00, 0x00,      // 7E ~
   0x1F, 0x1F, 0x1F, 0x1F, 0x1F, 0x1F, 0x1F,      // 7F block

#ifndef SDA5708_EXCLUDE_SPECIAL
   0x11, 0x0E, 0x11, 0x11, 0x1F, 0x11, 0x11,      // A umlaut
   0x11, 0x0E, 0x11, 0x11, 0x11, 0x11, 0x0E,      // O umlaut
   0x11, 0x00, 0x11, 0x11, 0x11, 0x11, 0x0E,      // U umlaut
   0x11, 0x00, 0x0E, 0x01, 0x0F, 0x11, 0x0F,      // a umlaut
   0x11, 0x00, 0x0E, 0x11, 0x11, 0x11, 0x0E,      // o umlaut
   0x11, 0x00, 0x11, 0x11, 0x11, 0x13, 0x0D,      // u umlaut
   0x0C, 0x12, 0x14, 0x12, 0x11, 0x16, 0x10,      // ß
   0x07, 0x08, 0x1E, 0x08, 0x1E, 0x08, 0x07,      // Euro
   0x02, 0x05, 0x02, 0x04, 0x07, 0x00, 0x00,      // power 2
   0x06, 0x01, 0x06, 0x01, 0x06, 0x00, 0x00,      // power 3
   0x02, 0x05, 0x02, 0x00, 0x00, 0x00, 0x00,      // degree	
   0x00, 0x00, 0x12, 0x12, 0x12, 0x1D, 0x10,      // micro
   0x00, 0x0E, 0x11, 0x11, 0x11, 0x0A, 0x1B,      // ohms
   0x00, 0x04, 0x0E, 0x04, 0x00, 0x0E, 0x00,      // plusminus
   0x00, 0x00, 0x0A, 0x04, 0x0A, 0x00, 0x00,      // times
   0x00, 0x04, 0x00, 0x0E, 0x00, 0x04, 0x00,      // divide
#endif

#ifdef SDA5708_INCLUDE_ARROWS
   0x00, 0x04, 0x0C, 0x1F, 0x0C, 0x04, 0x00,      // arrow left
   0x00, 0x04, 0x06, 0x1F, 0x06, 0x04, 0x00,      // arrow right
   0x00, 0x04, 0x0E, 0x1F, 0x04, 0x04, 0x00,      // arrow up
   0x00, 0x04, 0x04, 0x1F, 0x0E, 0x04, 0x00,      // arrow down 
   0x00, 0x1E, 0x1C, 0x1C, 0x12, 0x01, 0x00,      // arrow left & up
   0x00, 0x0F, 0x07, 0x07, 0x09, 0x10, 0x00,      // arrow right & up
   0x00, 0x01, 0x12, 0x1C, 0x1C, 0x1E, 0x00,      // arrow left & down
   0x00, 0x10, 0x09, 0x07, 0x07, 0x0F, 0x00,      // arrow right & down
   0x00, 0x04, 0x08, 0x1F, 0x08, 0x04, 0x00,      // light arrow left
   0x00, 0x04, 0x02, 0x1F, 0x02, 0x04, 0x00,      // light arrow right
   0x00, 0x04, 0x0E, 0x15, 0x04, 0x04, 0x00,      // light arrow up
   0x00, 0x04, 0x04, 0x15, 0x0E, 0x04, 0x00,      // light arrow down 
   0x00, 0x1E, 0x18, 0x14, 0x12, 0x01, 0x00,      // light arrow left & up
   0x00, 0x0F, 0x03, 0x05, 0x09, 0x10, 0x00,      // light arrow right & up
   0x00, 0x01, 0x12, 0x14, 0x18, 0x1E, 0x00,      // light arrow left & down
   0x00, 0x10, 0x09, 0x05, 0x03, 0x0F, 0x00,      // light arrow right & down
#endif

#ifdef SDA5708_INCLUDE_HBLOCKS
   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,      // horizontal stripes - empty
   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x1F,      // row 1
   0x00, 0x00, 0x00, 0x00, 0x00, 0x1F, 0x1F,      // row 1+2
   0x00, 0x00, 0x00, 0x00, 0x1F, 0x1F, 0x1F,      // row 1-3
   0x00, 0x00, 0x00, 0x1F, 0x1F, 0x1F, 0x1F,      // row 1-4
   0x00, 0x00, 0x1F, 0x1F, 0x1F, 0x1F, 0x1F,      // row 1-5
   0x00, 0x1F, 0x1F, 0x1F, 0x1F, 0x1F, 0x1F,      // row 1-6
   0x1F, 0x1F, 0x1F, 0x1F, 0x1F, 0x1F, 0x1F,      // row 1-7
   0x1F, 0x1F, 0x1F, 0x1F, 0x1F, 0x1F, 0x1F,      // row 1-7
   0x1F, 0x1F, 0x1F, 0x1F, 0x1F, 0x1F, 0x00,      // row 2-7
   0x1F, 0x1F, 0x1F, 0x1F, 0x1F, 0x00, 0x00,      // row 3-7
   0x1F, 0x1F, 0x1F, 0x1F, 0x00, 0x00, 0x00,      // row 4-7
   0x1F, 0x1F, 0x1F, 0x00, 0x00, 0x00, 0x00,      // row 5-7
   0x1F, 0x1F, 0x00, 0x00, 0x00, 0x00, 0x00,      // row 6+7
   0x1F, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,      // row 7
   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,      // empty
#endif

#ifdef SDA5708_INCLUDE_VBLOCKS
   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,      // vertical stripes - empty
   0x10, 0x10, 0x10, 0x10, 0x10, 0x10, 0x10,      // column 1
   0x18, 0x18, 0x18, 0x18, 0x18, 0x18, 0x18,      // column 12
   0x1C, 0x1C, 0x1C, 0x1C, 0x1C, 0x1C, 0x1C,      // column 123
   0x1E, 0x1E, 0x1E, 0x1E, 0x1E, 0x1E, 0x1E,      // column 1234
   0x1F, 0x1F, 0x1F, 0x1F, 0x1F, 0x1F, 0x1F,      // column 12345
   0x0F, 0x0F, 0x0F, 0x0F, 0x0F, 0x0F, 0x0F,      // column  2345
   0x07, 0x07, 0x07, 0x07, 0x07, 0x07, 0x07,      // column   345
   0x03, 0x03, 0x03, 0x03, 0x03, 0x03, 0x03,      // column    45
   0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01,      // column     5
   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,      // empty
   0x11, 0x11, 0x11, 0x11, 0x11, 0x11, 0x11,      // column 1   5
   0x1B, 0x1B, 0x1B, 0x1B, 0x1B, 0x1B, 0x1B,      // column 12 45
   0x1F, 0x1F, 0x1F, 0x1F, 0x1F, 0x1F, 0x1F,      // column 12345 
   0x0E, 0x0E, 0x0E, 0x0E, 0x0E, 0x0E, 0x0E,      // column  234
   0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04,      // column   3
#endif

#ifdef SDA5708_INCLUDE_GRAPHICS
   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,      // horizontal stripes - empty
   0x1F, 0x00, 0x00, 0x00, 0x00, 0x00, 0x1F,      // rows 1     7
   0x1F, 0x1F, 0x00, 0x00, 0x00, 0x1F, 0x1F,      // rows 12   67
   0x1F, 0x1F, 0x1F, 0x00, 0x1F, 0x1F, 0x1F,      // rows 123 567
   0x1F, 0x1F, 0x1F, 0x1F, 0x1F, 0x1F, 0x1F,      // rows 1234567
   0x00, 0x1F, 0x1F, 0x1F, 0x1F, 0x1F, 0x00,      // rows  23456
   0x00, 0x00, 0x1F, 0x1F, 0x1F, 0x00, 0x00,      // rows   345
   0x00, 0x00, 0x00, 0x1F, 0x00, 0x00, 0x00,      // rows    4
   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,      // empty
   0x00, 0x00, 0x04, 0x04, 0x04, 0x00, 0x00,      // special 1
   0x00, 0x0E, 0x0E, 0x0E, 0x0E, 0x0E, 0x00,      // special 2
   0x1F, 0x1F, 0x1F, 0x1F, 0x1F, 0x1F, 0x1F,      // special 3
   0x1F, 0x1F, 0x1B, 0x1B, 0x1B, 0x1F, 0x1F,      // special 4
   0x1F, 0x11, 0x11, 0x11, 0x11, 0x11, 0x1F,      // special 5 
   0x15, 0x0A, 0x15, 0x0A, 0x15, 0x0A, 0x15,      // checkerboard1
   0x0A, 0x15, 0x0A, 0x15, 0x0A, 0x15, 0x0A,      // checkerboard2
#endif

};

#endif // include
